Removed non-functional custom taxonomies meta boxes.

jorgefilipecosta opened the pull request:
These meta boxes don't work at all, and the new editor already shows panels to allow the use of custom taxonomies. So showing this meta boxes is a bug.

## How Has This Been Tested?
Add a custom taxonomy with a custom slug, verify a meta box with just the taxonomy title appear under Extended settings on the master branch, and does not appear on this branch.

jorgefilipecosta committed:
Removed non-functional custom taxonomies meta boxes.

These meta boxes don't work at all, and the new editor already shows panel to allow the use of custom meta boxes. So showing this meta boxes is a bug.

aduth reviewed (changes requested):
This looks to be a good improvement, but I'm still seeing a meta box for my custom taxonomy:

![Genre](https://user-images.githubusercontent.com/1779930/36755831-5e2550f0-1bdb-11e8-9a05-178458f96d12.png)

Entirety of the plugin:

```php
<?php

/**
 * Plugin Name: Demo CPT
 */

add_action( 'init', function() {
	register_post_type( 'book', [
		'label' => 'Book',
		'show_in_rest' => true,
		'public' => true,
		'show_ui' => true,
		'supports' => [ 'title', 'editor' ],
	] );

	register_taxonomy( 'genre', 'book', [
		'label' => __( 'Genre' ),
		'rewrite' => array( 'slug' => 'genre' ),
		'hierarchical' => false,
		'show_ui' => true, 
		'show_in_rest' => true,
	] );
} );
```

jorgefilipecosta commented:
Hi @aduth, it looks like the way to build the meta boxes ids is different for non-hierarchical taxonomies I updated the code to handle this case. I also removed the 'categorydiv' and 'tagsdiv-post_tag' from the hardcoded list as this cases are handled by the new code. Thank you for catching this case 👍

aduth commented on lib/meta-box-partial-page.php:
```diff
@@ -90,12 +90,23 @@ function gutenberg_filter_meta_boxes( $meta_boxes ) {
+	);
+
+	foreach ( $custom_taxonomies as $custom_taxonomy ) {
+		$core_side_meta_boxes [] = $custom_taxonomy->hierarchical ?
```
Might be worth a mentioned to where this is generated in the base core code:

https://github.com/WordPress/wordpress-develop/blob/c896326/src/wp-admin/edit-form-advanced.php#L288-L292

aduth reviewed (approved):
Works great 👍
